pkg/cloudprovider/provider/kubevirt/types/types.go

```diff
@@ -71,6 +72,20 @@ type Disk struct {
 	StorageClassName providerconfigtypes.ConfigVarString `json:"storageClassName,omitempty"`
 }
 
+// Affinity
+type Affinity struct {
+	PodAffinityPreset     providerconfigtypes.ConfigVarString `json:"podAffinityPreset,omitempty"`
```

Comment:
It might not be the best name ever... I did not want to go with `PodAffinity` as it's not the whole affinity definition.
It's just a kind of pre-selection of a pod affinity, containing only `hard` or `soft` or `` that leads to a pre-defined affinity based on the hostname and the vm label selector.

What would you suggest? Any help is welcome !

Reply:
Ah I see, I didn't really realize the switch statement in `getAffinity`. Then I guess it's okay.